client/state/comments/reducer.js

```diff
@@ -191,6 +192,38 @@ export const totalCommentsCount = createReducer(
 );
 
 /**
+ * Stores a site's comments tree.
+ * @param {Object} state Redux state (`comments.trees`)
+ * @param {Object} action Redux action
+ * @returns {Object} New Redux state.
+ */
+export const trees = ( state = {}, { siteId, status, tree, type } ) => COMMENTS_TREE_RECEIVE === type
+	? {
+		...state,
+		[ siteId ]: {
+			...state[ siteId ],
+			[ status ]: tree,
+		}
+	}
```

Comment:
we should ask ourselves if we also want post information. that's _relatively_ cheap on this API call and could make for some quicker transitions into the discussions for a given post.